app/assets/javascripts/admin/products/services/variant_unit_manager.js.coffee

```diff
@@ -56,10 +56,9 @@ angular.module("admin.products").factory "VariantUnitManager", (availableUnits)
 
     @unitScales: (unitType, availableUnits = null) ->
       scales = Object.keys(@units[unitType])
-      units = @units
       if availableUnits
         scales = scales.filter (scale) ->
-          availableUnits.includes(units[unitType][scale]['name'])
+          availableUnits.includes(VariantUnitManager.getUnitName(scale, unitType))
```

Comment:
I'd break this in two lines. One dot per line or one method call per line are good rules I think.

```ruby
unitName = VariantUnitManager.getUnitName(scale, unitType)
availableUnits.includes(unitName)
```

This is just my opinion.